src/test/java/com/google/jenkins/plugins/computeengine/integration/ComputeEngineCloudIgnoreProxyIT.java

```diff
@@ -0,0 +1,133 @@
+import static com.google.jenkins.plugins.computeengine.integration.ITUtil.initCredentials;
+import static com.google.jenkins.plugins.computeengine.integration.ITUtil.instanceConfigurationBuilder;
+import static com.google.jenkins.plugins.computeengine.integration.ITUtil.teardownResources;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import com.google.cloud.graphite.platforms.plugin.client.ComputeClient;
+import com.google.common.collect.ImmutableList;
+import com.google.jenkins.plugins.computeengine.ComputeEngineCloud;
+import com.google.jenkins.plugins.computeengine.ComputeEngineInstance;
+import com.google.jenkins.plugins.computeengine.InstanceConfiguration;
+import hudson.ProxyConfiguration;
+import hudson.model.FreeStyleBuild;
+import hudson.model.FreeStyleProject;
+import hudson.model.Node;
+import hudson.model.labels.LabelAtom;
+import hudson.tasks.Builder;
+import hudson.tasks.Shell;
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import jenkins.model.Jenkins;
+import lombok.extern.java.Log;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import org.jvnet.hudson.test.JenkinsRule;
+
+/**
+ * Integration test suite for {@link ComputeEngineCloud}. Verifies that when the ignore proxy flag
+ * is set to true, connections to GCE VMs bypass the proxy, and when set to false the proxy is used.
+ */
+@Log
+public class ComputeEngineCloudIgnoreProxyIT {
+  @ClassRule
+  public static Timeout timeout = new Timeout(5 * TEST_TIMEOUT_MULTIPLIER, TimeUnit.MINUTES);
+
+  @ClassRule public static JenkinsRule jenkinsRule = new JenkinsRule();
+
+  private static ComputeClient client;
+  private static Map<String, String> label = getLabel(ComputeEngineCloudIgnoreProxyIT.class);
+  private static ComputeEngineCloud cloud;
+  private static FreeStyleProject project;
+  private static Node worker;
+
+  @BeforeClass
+  public static void init() throws Exception {
+    log.info("init");
+    initCredentials(jenkinsRule);
+    cloud = initCloud(jenkinsRule);
+    client = initClient(jenkinsRule, label, log);
+
+    InstanceConfiguration instanceConfiguration =
+        instanceConfigurationBuilder()
+            .startupScript(DEB_JAVA_STARTUP_SCRIPT)
+            .numExecutorsStr(NUM_EXECUTORS)
+            .labels(LABEL)
+            .template(NULL_TEMPLATE)
+            .googleLabels(label)
+            .oneShot(true)
+            .ignoreProxy(true)
+            .build();
+
+    Jenkins jenkins = jenkinsRule.getInstance();
+    jenkins.proxy = new ProxyConfiguration("127.0.0.1", 8080);
+    jenkins.proxy.save();
+    jenkins.save();
+    cloud.setConfigurations(ImmutableList.of(instanceConfiguration));
+    assertTrue(
```

Comment:
Seems like 99-111 is the actual test. Consider moving this part down into a test.